### ISSUE TITLE
GSoC 2021 Proposal - BookReader Accessibility and Mobile Improvements

### ISSUE THREAD

**Issue by annsudhart** [internetarchive/bookreader#652]
# An Online Reading Experience for All
## GSoC 2021 Proposal - BookReader Accessibility and Mobile Improvements

TLDR: Hello, my name is Andrea, a student at the University of California San Diego. I believe working on the accessibility of BookReader can be helpful for its users, and want to work on improving it for GSOC 2021. If there are other things on BookReader I am open to hear what other ideas exist! 

I hope I get my proposal accepted. I went through the codebase and hope to prove myself by improving the testing suite (it seems like a pertinent issue based on the GitHub repo) and solving issues.

Some questions:
- Is this something that would be of interest to the BookReader team, and to anyone particular that could be a mentor? Or would a project like this be better for the IAUX Repo?
- Are there other accessibility-related features that could be of interest?
- Would creating tests or commit hooks to maintain accessibility be possible?

## Contents
- Summary
- Why Accessibility? How does this benefit BookReader?
- About me
- Timeframe

### Summary
Taken from [the Google Summer of Code guidelines](https://github.com/internetarchive/openlibrary/wiki/Google-Summer-of-Code).

1. What is BookReader?

> BookReader is an open source web application by the Internet Archive that serves as a reader for scanned books and documents. With over [3 Million books available](https://openlibrary.org/subjects/accessible_book), it allows anyone with an Internet connection to read an available book, whether embedded in a website, or found on Open Library.

2. Why is BookReader important?

> We need an accessible format for reading books online. With BookReader, we do not have to download an app, or jump through hurdles to access a book.

3.  What Opportunity exists for BookReader

> BookReader has made great progress in becoming more accessible with read-aloud, vision filter plugins, and keyboard commands for zooming and flipping pages. However, many of these great plugins are inaccessible by keyboard, and fall short of several WCAG guidelines when running an audit using Chrome's Lighthouse and Firefox's Accessibility DevTools.

4. Why now? Why is this opportunity possible now (and wasn’t possible before?)

> BookReader is constantly under development, and with more features coming over time there are more features that need to be considered for accessibility. Now would be a good time to close loose ends in accessibility and integrate accessibility considerations into the development pipeline. 

> I also checked previous iterations of GSOC and accessibility projects, and was impressed with the visual impairment feature by Giacomo Cognini three years ago. Would Drini Cami or someone on the team be interested in working on this with me, and discussing more about ways to improve accessibility? 

5. What does the solution look like?

> Widgets/buttons highlighted in a logical order while keyboard navigating, proper labeling of HTML elements. Furthermore, we can establish a way to maintain good WCAG accessibility as more code is added to BookReader. Like a custom [a11y check](https://www.npmjs.com/package/eslint-plugin-jsx-a11y)

6. What does success for BookReader look like? What is the potential impact, who will benefit?

> An easy way to navigate by keyboard and screen-reader, and accessibility tests/git commit hooks if applicable that will ensure accessibility standards are retained.

### Why Accessibility?

> As the Internet Archive strives to be accessible worldwide, we should also have high standards for visual impairments and keyboard accessibility, as improving accessibility is an improvement not only for those with impairments, but for everyone! I am passionate about accessibility, and believe BookReader should be accessible to everyone too.

### About me
I am Andrea Sudharta, and I major in Math-CS and have a minor in Cognitive Science. My interests lie within web devlopment and UX/UI. On the side I am a hackathon enthusiast and member of the student committee of the famous Geisel Library on campus. The BookReader project interested me because of my passion for open access to literature, accessibility, and realizing quality UI/UX with code. 

- [g]mail: sudharta.andrea
- Website: https://annsudhart.github.io

### Timeframe
- [Week 1 - 3] Keyboard navigation accessibility improvements for navbar, plugins, menu, BookReader embed
- [Week 4 -5] ARIA Labels for better element labeling for screen readers
- [Week 6 - 10] Establishing a framework for auditing for accessibility for future commits

**Comment by cdrini** [comment]
Hi, these are best submitted through the gsoc site. Thank you for your proposal!
